### DISCUSSION TITLE
Cann't fly stable through setpoint.

### DISCUSSION THREAD

**Discussion by ChenMathilda** [bitcraze/discussions#430] [Q&A]
Hi,
I try to use `setHoverSetpoint `to control UAV flight without lighthouse. However, I often encounter the problem that UAV turning over after takeoff. After many tests, I found that this problem occurs when the battery power is lower than 3.7v.
However, sometimes, when the battery is fully charged, the UAV will also turn over after takeoff. 
I want to know if anyone has the same problem as me, or there were other reasons I haven't noticed?

**Comment by knmcguire** [comment]
It depends on how high the setpoint is, but this seems that the motors are perhaps old and get oversaturated too much and can't handle the stability.

Couple of things you can do:
1- Make sure the crazyflie turns it's propellors a bit before takeoff
2- Use the [highlevel commander to take off](https://www.bitcraze.io/documentation/repository/crazyflie-lib-python/master/api/cflib/crazyflie/high_level_commander/#:~:text=def%20takeoff(self%2C%20absolute_height_m%2C%20duration_s%2C%20group_mask%3D0%2C%20yaw%3D0.0)), this has a more smooth take off trajectory that make it less likely for the motors to oversaturate
3- Buy a thrust upgrade kit to upgrade the motors and propellers which should be able to handle such a sharp takeoff setpoint: https://store.bitcraze.io/collections/spare-parts/products/thrust-upgrade-bundle-for-crazyflie-2-x
